data/spells/scripts/support/findPerson.lua

```diff
@@ -0,0 +1,129 @@
+local directionNames = {
+	DIRECTION_NORTH = "north.",
+	DIRECTION_SOUTH = "south.",
+	DIRECTION_EAST = "east.",
+	DIRECTION_WEST = "west.",
+	DIRECTION_NORTHEAST = "north-east.",
+	DIRECTION_NORTHWEST = "north-west.",
+	DIRECTION_SOUTHEAST = "south-east.",
+	DIRECTION_SOUTHWEST = "south-west."
+}
+
+local LEVEL_LOWER = 1
+local LEVEL_SAME = 2
+local LEVEL_HIGHER = 3
+
+local DISTANCE_BESIDE = 1
+local DISTANCE_CLOSE = 2
+local DISTANCE_FAR = 3
+local DISTANCE_VERYFAR = 4
+
+function onCastSpell(creature, var)
+	if not creature:getPlayer() then
+		return false
+	end
+
+	local targetPlayer = Player(Variant.getString(var))
+	if not targetPlayer then
+		return false
+	end
+
+	if targetPlayer:getGroup():getAccess() and not creature:getGroup():getAccess() then
+		creature:sendCancelMessage(RETURNVALUE_PLAYERWITHTHISNAMEISNOTONLINE)
+		creature:getPosition():sendMagicEffect(CONST_ME_POFF)
+		return false
+	end
+
+	local playerPosition = creature:getPosition()
+	local targetPosition = targetPlayer:getPosition()
+	local distance = {
+		x = playerPosition.x - targetPosition.x,
+		y = playerPosition.y - targetPosition.y,
+		z = playerPosition.z - targetPosition.z
+	}
+
+	local level = LEVEL_SAME
+	if distance.z > 0 then
+		level = LEVEL_HIGHER
+	elseif distance.z < 0 then
+		level = LEVEL_LOWER
+	end
+
+	local distanceOutput = DISTANCE_VERYFAR
+	if math.abs(distance.x) < 4 and math.abs(distance.y) < 4 then
+		distanceOutput = DISTANCE_BESIDE
+	else
+		local distanceFormula = ((distance.x * distance.x) + (distance.y * distance.y))
+		if distanceFormula < 1000 then
+			distanceOutput = DISTANCE_CLOSE
+		elseif distanceFormula < 75076 then
```

Comment:
You can change this 75076 with 274 * 274 so it doesn't look arbitrary. You could even map to a `FAR_DISTANCE_RADIUS` or so to give meaning.